### PR TITLE
Handle map wide events during replay

### PR DESCRIPTION
**Pull request by ahmetmircik**
second part of the fix: https://github.com/hazelcast/hazelcast/pull/12664

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/operation/PublisherCreateOperation.java [hunk 1]
```diff
@@ -165,42 +162,77 @@ private QueryResult runInitialQuery() {
...
                 continue;
             }
-            Object toObject = mapServiceContext.toObject(result);
-            List<QueryCacheEventData> eventDataList = (List<QueryCacheEventData>) toObject;
+            eventsInOneAcc = mapServiceContext.toObject(eventsInOneAcc);
+            List<QueryCacheEventData> eventDataList = (List<QueryCacheEventData>) eventsInOneAcc;
             for (QueryCacheEventData eventData : eventDataList) {
-                QueryResultRow entry = createQueryResultEntry(eventData);
-                add(queryResult, entry);
+                if (eventData.getDataKey() == null) {
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/operation/PublisherCreateOperation.java]: this is the main fix here, before we were not handling clear-all, evict-all events at this point.

File: hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/operation/PublisherCreateOperation.java [hunk 2]
```diff
@@ -228,32 +260,7 @@ private QueryResultRow createQueryResultEntry(QueryCacheEventData eventData) {
         return QueryCacheUtil.getAccumulators(context, mapName, cacheId).keySet();
     }
 
-    /**
-     * Reads the accumulator in a partition.
-     */
-    private final class PartitionCallable implements Callable<Object> {
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/operation/PublisherCreateOperation.java]: this was unneeded and i removed it.

File: hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/operation/PublisherCreateOperation.java [hunk 3]
```diff
@@ -165,42 +162,77 @@ private QueryResult runInitialQuery() {
...
+     * received events during execution of the initial query.
      */
-    private void replayEventsOnResultSet(final QueryResult queryResult) throws Exception {
-        Collection<Object> resultCollection = readAccumulators();
-        for (Object result : resultCollection) {
-            if (result == null) {
+    private void replayEventsOverResultSet(QueryResult queryResult) throws Exception {
+        Map<Integer, Future<Object>> future = readAccumulators();
+        for (Map.Entry<Integer, Future<Object>> entry : future.entrySet()) {
+            Integer partitionId = entry.getKey();
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/operation/PublisherCreateOperation.java]: Can this ever be `null`? Due to the used `Int2ObjectHashMap` I guess not. I would then use `int`, so it's clear that there are no `null` checks needed.

File: hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/operation/PublisherCreateOperation.java [hunk 4]
```diff
@@ -165,42 +162,77 @@ private QueryResult runInitialQuery() {
...
             }
         }
     }
 
-    private Collection<Object> readAccumulators() {
+    /**
+     * Remove matching entries from given result set with the given
+     * partition ID.
+     */
+    private void removePartitionResults(QueryResult queryResult, Integer partitionId) {
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/operation/PublisherCreateOperation.java]: Same here, an `Integer partitionId` looks strange. There are more candidates to check in this class.

File: hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/operation/PublisherCreateOperation.java (hunk 3 above)
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/operation/PublisherCreateOperation.java]: done

File: hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/operation/PublisherCreateOperation.java (hunk 4 above)
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/operation/PublisherCreateOperation.java]: done

**Pull request merged**